src/05-derivatives/05-diffusion-derivatives.md

```diff
@@ -18,132 +18,359 @@ should be included in the pipeline documentation.
+| GradientNonLinearityGeometryCorrection | OPTIONAL. Boolean. Correction for geometric distortions arising from non-linearity of gradients                                                   |
+| GradientNonLinearityQSpaceCorrection   | OPTIONAL. Boolean. Correction for spatial inhomogeneity of diffusion sensitisation gradient strength                                              |
+| SliceDropoutDetection                  | OPTIONAL. Boolean. Detection of signal dropout in acquired slices during pre-processing                                                           |
+| SliceDropoutReplacement                | OPTIONAL. Boolean. Replacement of image data within slices containing signal dropout with predicted values                                        |
+| BiasFieldCorrectionMethod              | OPTIONAL. String. Method (if any) used for correction of B1 RF field inhomogeneity                                                                |
 
-## Diffusion Models (and input parameters)
+## Diffusion models
 
-Diffusion MRI can be modeled using various paradigms to extract a more easily
-understandable representation of the diffusion process and the underlying
-structure. To do so, parameters might be needed to control how the signal is fit
-to the model. Those parameters are called **input parameters** in the following.
-Once the model is fit, the resulting representation can be saved using a number
-of values per voxel. Those values will be called **output** or **estimated
-parameters** in the following.
+Diffusion MRI can be modeled using various paradigms to extract more
+informative representations of the diffusion process and the underlying
+biological structure. A wide range of such models are available, each of
+which has its own unique requirements with respect to:
 
-**Estimated parameters** are saved as NIFTI files (see section Models for the
-expected content of each model), and **input parameters** are saved in the
-sidecar JSON file.
+-   The input parameters required to define / constrain the model;
 
-The following is a general example of naming convention:
+-   The appropriate data representation in which the resulting
+    parameters estimated of / from the model are stored on the filesystem;
+
+-   The requirements for encapsulation and complete representation of
+    derived orientation information, which is a key strength of diffusion
+    MRI but presents unique challenges for correct interpretation.
+
+### Parameter terminology
+
+1.  *Intrinsic* parameter: Item that is the direct result of fitting
+    the diffusion model to the empirical diffusion-weighted data.
+
+1.  *Input* parameter: Item that influences the conformation of the
+    diffusion model to the empirical diffusion-weighted data.
+
+1.  *Extrinsic* parameter: Item that can be calculated from prior
+    estimated intrinsic model parameters, without necessitating reference
+    to the empirical diffusion-weighted data.
+
+### File names

```

Comment:
I struggled with the order of sections as well. I think it's difficult to move "intrinsic model parameters" above the "data representations" section, since the former contains so many links to the latter. The only thing in between them then is "orientation specification": while it would be 'safe' to have this below "intrinsic model parameters", it itself is reasonably tightly linked to "data representations". This would also move "intrinsic model parameters" further away from "input parameters" and "extrinsic model parameters"; which are we've seen from this process are quite strongly related also. I understand the motivation to get the model definitions up as high as possible, I experienced the same thing.

I think if this were to change, it would not be "moving the table of parameters up" so much as "moving data representations & orientation specification to the end (just before the examples)".

So instead of:

Diffusion derivatives
- Preprocessed diffusion-weighted images
- Diffusion models
  - Parameter terminology
  - File names
  - Data representations
  - Orientation specification
    - Spherical Harmonics bases
  - Intrinsic model parameters
    - Model bootstrapping
  - Input model parameters
  - Extrinsic model parameters
- Demonstrative examples

, it would instead be:

Diffusion derivatives
- Preprocessed diffusion-weighted images
- Diffusion models
  - Parameter terminology
  - File names
  - Intrinsic model parameters
    - Model bootstrapping
  - Input model parameters
  - Extrinsic model parameters
  - Data representations
  - Orientation specification
    - Spherical Harmonics bases
- Demonstrative examples

It would mean that "data representations" would get referred to *a lot* before actually appearing though.

Reply:
Looking at the proposed change itself: I do think it would be useful to have a brief introduction paragraph at the head of some of the sections. But in looking to adopt / modify your suggestion here, I'm still not happy with the overall "file names" section; I also think some of what you were starting to get at in that text was not wholly specific to "file names". So what I might do instead is eventually decide on a different structure for what's currently the "file names" section, as the requisite modifications from that point may be entirely different.

Reply:
See 8b27828 where "orientation specification" is moved down, concomitantly moving the parameter tables up.